doc/input.md

```diff
@@ -1,41 +1,44 @@
-
-For example, to add a tap listener ("on click"):
-
-```dart
-    Flame.util.addGestureRecognizer(new TapGestureRecognizer()
-        ..onTapDown = (TapDownDetails evt) => game.handleInput(evt.globalPosition.dx, evt.globalPosition.dy));
-```
-
-Where `game` is a reference to your game object and `handleInput` is a method you create to handle the input inside your game.
-
-If your game doesn't have other screens, just call this after your `runApp` call, in the `main` method.
-
-Here are some example of more complex Gesture Recognizers:
+The `Game` class has a vast number of methods for handling touch controls, to use it, just override the method you want to use, and that method will start receiving events, below there is the whole list of available methods:
 
 ```dart
-    MyGame() {
-        // other init...
-
-        Flame.util.addGestureRecognizer(createDragRecognizer());
-        Flame.util.addGestureRecognizer(createTapRecognizer());
-    }
-
-    GestureRecognizer createDragRecognizer() {
-        return new ImmediateMultiDragGestureRecognizer()
-            ..onStart = (Offset position) => this.handleDrag(position);
-    }
-
-    TapGestureRecognizer createTapRecognizer() {
-        return new TapGestureRecognizer()
-            ..onTapUp = (TapUpDetails details) => this.handleTap(details.globalPosition);;
-    }
+  void onTap() {}
+  void onTapCancel() {}
+  void onTapDown(TapDownDetails details) {}
+  void onTapUp(TapUpDetails details) {}
+  void onSecondaryTapDown(TapDownDetails details) {}
+  void onSecondaryTapUp(TapUpDetails details) {}
+  void onSecondaryTapCancel() {}
+  void onDoubleTap() {}
+  void onLongPress() {}
+  void onLongPressStart(LongPressStartDetails details) {}
+  void onLongPressMoveUpdate(LongPressMoveUpdateDetails details) {}
+  void onLongPressUp() {}
+  void onLongPressEnd(LongPressEndDetails details) {}
+  void onVerticalDragDown(DragDownDetails details) {}
+  void onVerticalDragStart(DragStartDetails details) {}
+  void onVerticalDragUpdate(DragUpdateDetails details) {}
+  void onVerticalDragEnd(DragEndDetails details) {}
+  void onVerticalDragCancel() {}
+  void onHorizontalDragDown(DragDownDetails details) {}
+  void onHorizontalDragStart(DragStartDetails details) {}
+  void onHorizontalDragUpdate(DragUpdateDetails details) {}
+  void onHorizontalDragEnd(DragEndDetails details) {}
+  void onHorizontalDragCancel() {}
+  void onForcePressStart(ForcePressDetails details) {}
+  void onForcePressPeak(ForcePressDetails details) {}
+  void onForcePressUpdate(ForcePressDetails details) {}
+  void onForcePressEnd(ForcePressDetails details) {}
+  void onPanDown(DragDownDetails details) {}
+  void onPanStart(DragStartDetails details) {}
+  void onPanUpdate(DragUpdateDetails details) {}
+  void onPanEnd(DragEndDetails details) {}
+  void onPanCancel() {}
+  void onScaleStart(ScaleStartDetails details) {}
+  void onScaleUpdate(ScaleUpdateDetails details) {}
+  void onScaleEnd(ScaleEndDetails details) {}
 ```
```

Comment:
mention that those are direct ported from the actual flutter methods and add the link we had previously where you can find more details for each one

also, add an example of simple tap/click handler